pkg/kudoctl/packages/writer/writer_tar.go

```diff
@@ -112,5 +112,9 @@ func TgzDir(fs afero.Fs, path string, w io.Writer) (err error) {
 
 // packageVersionedName provides the version name of a package provided a set of Files.  Ex. "zookeeper-0.1.0"
 func packageVersionedName(pkg *packages.Files) string {
-	return fmt.Sprintf("%v-%v", pkg.Operator.Name, pkg.Operator.Version)
+	if pkg.Operator.AppVersion == "" {
+		return fmt.Sprintf("%v-%v", pkg.Operator.Name, pkg.Operator.OperatorVersion)
+	}
+
+	return fmt.Sprintf("%v-%v-%v", pkg.Operator.Name, pkg.Operator.AppVersion, pkg.Operator.OperatorVersion)
```

Comment:
I'd use `_` to separate both versions since `-` is valid within semver.
```suggestion
	return fmt.Sprintf("%v-%v_%v", pkg.Operator.Name, pkg.Operator.AppVersion, pkg.Operator.OperatorVersion)
```